src/Grpc.Net.Client/Balancer/Internal/SocketConnectivitySubchannelTransport.cs

```diff
@@ -300,6 +309,7 @@ public async ValueTask<Stream> GetStreamAsync(BalancerAddress address, Cancellat
         lock (Lock)
         {
             _activeStreams.Add(new ActiveStream(address, socket, stream));
+            SocketConnectivitySubchannelTransportLog.StreamCreated(_logger, _subchannel.Id, address, _activeStreams.Count);
```

Comment:
logging inside the lock troubles me; is this intentional? would it perhaps be better to snag the `Count` inside the lock and log outside?

Reply:
I don't think this is a concern. There is logging inside locks throughout gRPC and ASP.NET Core.